pkg/controller/jobframework/setup.go

```diff
@@ -92,6 +92,11 @@ func (m *integrationManager) setupControllers(ctx context.Context, mgr ctrl.Mana
 						log.Error(err, "Failed to setup controller and webhook for job framework")
 					}
 				})
+				// Webhook must be registered now.
```

Comment:
Actually, maybe it is safer to register before spinning the waitForAPI goroutine to avoid the two calls to `SetupWebhook` racing. I bet there is a mutex inside the controller-runtime, but it seems just safer. wdyt?

Reply:
Done.